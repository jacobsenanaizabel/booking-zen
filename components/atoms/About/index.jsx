import React from 'react';
import PropTypes from 'prop-types';
//import { Container, ButtonStyled } from './styles';

const About = () => {

  return (
    <></>
  );
};

About.propTypes = {
  errorHandler: PropTypes.func,
};

About.defaultProps = {
  errorHandler: null,
};
export default About;
