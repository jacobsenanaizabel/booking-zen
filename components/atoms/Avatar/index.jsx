import React from 'react';
import PropTypes from 'prop-types';
import { ContainerAvatar, User } from './styles';

const Avatar = ({img}) => {

  return (
    <>
      <ContainerAvatar>
        <figure>
          <User src={img}/>
        </figure>
      </ContainerAvatar>
    </>
  );
};

Avatar.propTypes = {
  img: PropTypes.string,
};

export default Avatar;
