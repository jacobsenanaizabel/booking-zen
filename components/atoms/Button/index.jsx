import React from 'react';
import PropTypes from 'prop-types';
import { Container, ButtonStyled } from './styles';

const Button = ({text, ...other}) => {

  return (
    <Container>
      <ButtonStyled
        tabindex="0"
        role="button"
        {...other}
      >
       {text}
      </ButtonStyled>
    </Container>
  );
};

Button.propTypes = {
  text: PropTypes.string,
};

Button.defaultProps = {
  text: 'OK'
};
export default Button;
