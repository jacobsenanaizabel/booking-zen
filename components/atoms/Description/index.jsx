import React from 'react';
import PropTypes from 'prop-types';
import { ContainerDescription } from './styles';
import {Text} from '../index'

const Description = ({text}) => {

  return (
    <>
    <ContainerDescription>
      <Text text={text} color="text" size="1rem"/>
    </ContainerDescription>
    </>
  );
};

Description.propTypes = {
  errorHandler: PropTypes.func,
};

Description.defaultProps = {
  errorHandler: null,
};
export default Description;
