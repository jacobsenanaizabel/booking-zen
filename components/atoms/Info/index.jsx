import React from 'react';

import PropTypes from 'prop-types';
import { Text } from '../index';
import { Info as About, ContainerInfo } from './styles';

const Info = ({location, profession}) => {   
  return (
    <>
    <ContainerInfo>
    <About>
      <Text text={profession} size="1rem" color="black" weight="700"/> 
      <Text text={'- '+location} size="1rem" color="purple" weight="700"/>
    </About>
  </ContainerInfo>
    </>
  );
};


Info.propTypes = {
  location: PropTypes.string,
  profession: PropTypes.string
};

export default Info;
