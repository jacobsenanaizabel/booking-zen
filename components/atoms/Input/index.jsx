import React from 'react';
import { InputStyled } from './styles';

const Input = ({placeholder, onChange}) => {
  return (
    <>
      <InputStyled placeholder={placeholder} onChange={onChange}/>
    </>
  );
}

Input.propTypes = {
};

Input.defaultProps = {
};

export default Input;


