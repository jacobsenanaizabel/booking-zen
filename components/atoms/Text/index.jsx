import React from 'react';
import PropTypes from 'prop-types';

import {
    Text as TextStyled
} from './styles';


const Title = ({text, size, color, weight, margin}) => {
  return (
    <>
        <TextStyled size={size} color={color} weight={weight} margin={margin} >{text}</TextStyled>
    </>
  );
};

Title.propTypes = {
  text: PropTypes.any
};


export default Title;