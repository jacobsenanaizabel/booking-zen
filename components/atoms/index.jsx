export { default as Title } from './Title';
export { default as Input } from './Input';
export { default as Button } from './Button';
export { default as About } from './About';
export { default as Avatar } from './Avatar';
export { default as Description } from './Description';
export { default as Price } from './Price';
export { default as Rating } from './Rating';
export { default as Text } from './Text';