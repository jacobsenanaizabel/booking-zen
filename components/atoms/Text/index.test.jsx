import React from 'react';
import { render } from "@testing-library/react";
import {Text} from '@components/atoms/index';

describe('Text tests', () => {

  const fakeText = "Testing text";

  it('should match snapshot with default Text', () => {

    const { asFragment } = render(<Text />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot with default Text with size', () => {

    const { asFragment } = render(<Text size="3rem" text={fakeText} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot with default Text with color', () => {
    const { container } = render(
        <>
          <Text color="text" text={fakeText}/>
          <Text color="violet" text={fakeText}/>
          <Text color="purple" text={fakeText}/>
          <Text color="black" text={fakeText}/>
          <Text color="border" text={fakeText}/>
        </>
      );
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with default Text with font-weight', () => {
    const { asFragment } = render(
        <>
          <Text weight="600" text={fakeText}/>
        </>
      );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot with default Text with margin', () => {
    const { asFragment } = render(
        <>
          <Text margin="0px" text={fakeText}/>
        </>
      );
    expect(asFragment()).toMatchSnapshot();
  });

});