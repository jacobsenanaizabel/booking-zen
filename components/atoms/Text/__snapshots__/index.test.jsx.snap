// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text tests should match snapshot with default Text 1`] = `
<DocumentFragment>
  <p
    class="styles__Text-sc-tpjft1-0 hvhVHE"
  />
</DocumentFragment>
`;

exports[`Text tests should match snapshot with default Text with color 1`] = `
<div>
  <p
    class="styles__Text-sc-tpjft1-0 erpmEK"
    color="text"
  >
    Testing text
  </p>
  <p
    class="styles__Text-sc-tpjft1-0 iRhMhG"
    color="violet"
  >
    Testing text
  </p>
  <p
    class="styles__Text-sc-tpjft1-0 cUvknR"
    color="purple"
  >
    Testing text
  </p>
  <p
    class="styles__Text-sc-tpjft1-0 hmGMUU"
    color="black"
  >
    Testing text
  </p>
  <p
    class="styles__Text-sc-tpjft1-0 kzzsuD"
    color="border"
  >
    Testing text
  </p>
</div>
`;

exports[`Text tests should match snapshot with default Text with font-weight 1`] = `
<DocumentFragment>
  <p
    class="styles__Text-sc-tpjft1-0 gSaWih"
  >
    Testing text
  </p>
</DocumentFragment>
`;

exports[`Text tests should match snapshot with default Text with margin 1`] = `
<DocumentFragment>
  <p
    class="styles__Text-sc-tpjft1-0 erxhS"
  >
    Testing text
  </p>
</DocumentFragment>
`;

exports[`Text tests should match snapshot with default Text with size 1`] = `
<DocumentFragment>
  <p
    class="styles__Text-sc-tpjft1-0 jxPzZx"
  >
    Testing text
  </p>
</DocumentFragment>
`;
